openedx/core/djangoapps/credit/signals.py

```diff
@@ -33,25 +32,6 @@ def on_course_publish(course_key):
         log.info(u'Added task to update credit requirements for course "%s" to the task queue', course_key)
 
 
-@receiver(SignalHandler.pre_publish)
-def on_pre_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
-    """
-    Create user partitions for verification checkpoints.
-
-    This is a pre-publish step since we need to write to the course descriptor.
-    """
-    from openedx.core.djangoapps.credit import api
-    if api.is_credit_course(course_key):
```

Comment:
Regarding the credit app, [it doesn't seem like there's a lot of development ongoing](https://github.com/edx/edx-platform/commits/master/openedx/core/djangoapps/credit), but I do feel like I'd rather do that removal in a separate ticket.

I also think it makes sense to treat these counts as an integration style test, so the increased query counts on this PR shouldn't be a blocker.